### PR TITLE
plugins: replace to_native(), to_text(), str() with str() where possible or leave it away in f-string formatting

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
With Python 3.6+ `to_native()` and `to_text()` are glorified `str()` calls **assuming they are not potentially decoding byte strings to unicode strings**. Remove them (in f-strings) or replace them with `str()` if possible.

##### ISSUE TYPE
- Refactoring Pull Request

##### COMPONENT NAME
plugins/action/shutdown.py
plugins/cache/redis.py
plugins/callback/logentries.py
plugins/callback/slack.py
plugins/connection/chroot.py
plugins/filter/from_csv.py
plugins/filter/from_ini.py
plugins/filter/to_ini.py
plugins/inventory/cobbler.py
plugins/inventory/gitlab_runners.py
plugins/inventory/iocage.py
plugins/inventory/lxd.py
plugins/inventory/nmap.py
plugins/inventory/opennebula.py
plugins/inventory/proxmox.py
plugins/inventory/scaleway.py
plugins/inventory/virtualbox.py
plugins/lookup/cyberarkpassword.py
plugins/lookup/dig.py
plugins/lookup/dnstxt.py
plugins/lookup/etcd3.py
plugins/lookup/lmdb_kv.py
plugins/lookup/manifold.py
plugins/lookup/onepassword.py
plugins/lookup/tss.py

### CODE REVIEW

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this for the next release.

File: plugins/inventory/iocage.py [hunk 1]
```diff
@@ -201,11 +201,10 @@ def get_inventory(self, path):
             try:
                 t_stdout = to_text(stdout, errors='surrogate_or_strict')
             except UnicodeError as e:
-                raise AnsibleError('Invalid (non unicode) input returned: %s' % to_native(e)) from e
+                raise AnsibleError(f'Invalid (non unicode) input returned: {e}') from e
 
         except Exception as e:
-            raise AnsibleParserError('Failed to parse %s: %s' %
-                                     (to_native(path), to_native(e))) from e
+            raise AnsibleParserError(f'Failed to parse {to_native(path)}: {e}') from e
```
Comment (russoz) [inline plugins/inventory/iocage.py:207]:
```suggestion
            raise AnsibleParserError(f'Failed to parse {path}: {e}') from e
```

File: plugins/inventory/iocage.py [hunk 2]
```diff
@@ -201,11 +201,10 @@ def get_inventory(self, path):
             try:
                 t_stdout = to_text(stdout, errors='surrogate_or_strict')
             except UnicodeError as e:
-                raise AnsibleError('Invalid (non unicode) input returned: %s' % to_native(e)) from e
+                raise AnsibleError(f'Invalid (non unicode) input returned: {e}') from e
```
Comment (russoz) [inline plugins/inventory/iocage.py:204]: Lines 198-199 above were clearly missed by the automated tool, they might be rewritten as an f-str and another `to_native()` would disappear.

File: plugins/inventory/lxd.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ def load_json_data(path):
             with open(path, 'r') as json_file:
                 return json.load(json_file)
         except (IOError, json.decoder.JSONDecodeError) as err:
-            raise AnsibleParserError(f'Could not load the test data from {to_native(path)}: {to_native(err)}')
+            raise AnsibleParserError(f'Could not load the test data from {to_native(path)}: {err}')
```
Comment (russoz) [inline plugins/inventory/lxd.py:214]:
```suggestion
            raise AnsibleParserError(f'Could not load the test data from {path}: {err}')
```

File: plugins/lookup/etcd3.py [hunk 4]
```diff
@@ -218,12 +218,12 @@ def run(self, terms, variables, **kwargs):
                         if val and meta:
                             ret.append({'key': to_native(meta.key), 'value': to_native(val)})
                 except Exception as exp:
-                    display.warning(f'Caught except during etcd3.get_prefix: {to_native(exp)}')
+                    display.warning(f'Caught except during etcd3.get_prefix: {exp}')
             else:
                 try:
                     val, meta = etcd.get(term)
                     if val and meta:
                         ret.append({'key': to_native(meta.key), 'value': to_native(val)})
```
Comment (russoz) [inline plugins/lookup/etcd3.py:226]: Maybe `str(meta.key)` and `str(val)` here as well?

Comment (russoz) [reply plugins/lookup/etcd3.py:226]: Same in line 219 above

File: plugins/lookup/lmdb_kv.py [hunk 5]
```diff
@@ -96,7 +96,7 @@ def run(self, terms, variables=None, **kwargs):
         try:
             env = lmdb.open(str(db), readonly=True)
         except Exception as e:
-            raise AnsibleError(f"LMDB can't open database {db}: {to_native(e)}")
+            raise AnsibleError(f"LMDB can't open database {db}: {e}")
```
Comment (russoz) [inline plugins/lookup/lmdb_kv.py:99]:
```suggestion
            raise AnsibleError(f"LMDB cannot open database {db}: {e}")
```

**Review by russoz — COMMENTED** [review]
Couple of adjustments, couple of maybes

File: plugins/inventory/iocage.py (hunk 1 above)
Comment (felixfontein) [reply plugins/inventory/iocage.py:207]: `path` could be a byte string, in which case the output would change from `Failed to parse /foo/` to `Failed to parse b'/foo/'`. (I didn't check whether it really could be a byte string, I only removed to_native/str/... from places where it was obviously safe...)

File: plugins/inventory/iocage.py (hunk 2 above)
Comment (felixfontein) [reply plugins/inventory/iocage.py:204]: The inventory plugin iocage was only merged after your tool ran :)

File: plugins/inventory/lxd.py (hunk 3 above)
Comment (felixfontein) [reply plugins/inventory/lxd.py:214]: Same as above.

File: plugins/lookup/etcd3.py (hunk 4 above)
Comment (felixfontein) [reply plugins/lookup/etcd3.py:226]: Same as above, `to_native()` also converts byte strings to unicode strings, which is very different from what `str()` does.

File: plugins/inventory/iocage.py (hunk 2 above)
Comment (felixfontein) [reply plugins/inventory/iocage.py:203]: 8353cb6967308f3d2d96c2b363dceae06d6384d4

**Review by russoz — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for reviewing this!

**Pull request merged**